pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -280,10 +347,11 @@ func (mc *multiCluster) StartCluster(
 	}
 
 	return &clusterInfo{
-		pgURL:        pgURL,
-		sysSQL:       sqlRunner,
-		db:           db,
-		gatewayNodes: nodes,
+		pgURL:  pgURL,
+		sysSQL: sqlRunner,
+		db:     db,
+		// Need to keep the node that we started the stream on
+		gatewayNodes: []int{node[0]},
```

Comment:
Michael and I talked about this in chat, I'll pick up https://github.com/cockroachdb/cockroach/pull/127613 before I can do this since it won't work currently.